tests/drum/unit/test_structured_input_read_utils.py

```diff
@@ -66,3 +68,12 @@ def test_read_structured_input_arrow_csv_na_consistency(self, tmp_path):
         # To do an exact comparison, compare None and np.nan "masks".
         assert_frame_equal(csv_df.applymap(is_nan), arrow_df.applymap(is_nan))
         assert_frame_equal(csv_df.applymap(is_none), arrow_df.applymap(is_none))
+
+    def test_read_structured_input_csv_unicode_error_handled(self):
+        data = "a\nb\nc\nd\ne\nf\n"
+        tmp_file = tempfile.NamedTemporaryFile(suffix=".csv")
+        with open(tmp_file.name, "w", encoding="utf-16") as f:
+            f.write(data)
+        with pytest.raises(DrumCommonException):
```

Comment:
pytest suggestion if you use the suggested text:
```suggestion
        with pytest.raises(DrumCommonException, match="Supplied CSV input file encoding must be UTF-8."):
```